### PR TITLE
Envelopes

### PR DESCRIPTION
**Pull request by Shadowmech88**
empty_envelope demo_envelope full_envelope demo_envelope_full full_envelope_torn
Adds envelopes.
Envelopes can be folded out of sheets of paper. Unlike other folded items, they cannot be unfolded.
Envelopes can be stamped, written on, and read just like paper.
Envelopes can have a sheet of paper placed inside them, the paper can be removed via Alt+click.
Using the envelope will seal it, if open, or tear it open, if sealed. An envelope can also be opened by using a sharp bladed item on it. Once opened, an envelope cannot be resealed.

Envelopes can be used to ensure tamper-free and spying-free communication, or at least expose those who do so, as the paper inside cannot be read or altered without irreversibly and obviously damaging the envelope.

:cl:
 * rscadd: Added envelopes, which can be folded out of sheets of paper. Envelopes can be stamped, written on, and read, and can have sheets of paper placed into them. The paper can be removed via Alt+click.

### CODE REVIEW

**Comment by DamianX** [comment]
I love it, but I'm not too fond of the alt-click to remove the paper. Perhaps holding it in one hand while interacting with it using the other hand is better? Like pizza boxes, for instance. I don't know which is more intuitive.

File: code/modules/paperwork/envelope.dm [hunk 1]
```diff
@@ -0,0 +1,81 @@
+/obj/item/weapon/paper/envelope
+	name = "envelope"
+	icon_state = "envelope_open"
+
+	var/obj/item/weapon/paper/contained_paper
+	var/open = TRUE
+	var/torn = FALSE
+	var/nano
```
Comment (PJB3005) [inline code/modules/paperwork/envelope.dm]: What's this?

Comment (Shadowmech88) [reply code/modules/paperwork/envelope.dm]: Paper folding code assumes that every result is of the `/obj/item/weapon/p_folded` type, which has a `nano` var, which paper folding code accesses.
Adding a var to envelopes with that name makes the folding code *just werk* despite them not being a subtype of `p_folded`.

I could rewrite paper folding code instead if you'd like.

Comment (PJB3005) [reply code/modules/paperwork/envelope.dm]: Dear god that's awful. Rewrite please.

**Comment by Megamaw** [comment]
alt-click is more intuitive in my opinion. it makes it harder to accidentally open.

**Comment by Shadowmech88** [comment]
You open the envelope via `attack_self()`. Alt+click is just for removing the contents of an unsealed/opened envelope.

**Comment by Blithering** [comment]
Can they be set on fire with lighters? The first thing I thought to write on one was 'burn after reading'

**Comment by Shadowmech88** [comment]
They are children of paper, and so have all the behaviors that paper has.

**Pull request merged**